tests/realtikvtest/pessimistictest/pessimistic_test.go

```diff
@@ -2816,6 +2817,66 @@ func TestAsyncCommitCalTSFail(t *testing.T) {
 	tk2.MustExec("commit")
 }
 
+func TestAsyncCommitAndForeignKey(t *testing.T) {
+	defer config.RestoreFunc()()
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.TiKVClient.AsyncCommit.SafeWindow = time.Second
+		conf.TiKVClient.AsyncCommit.AllowedClockDrift = 0
+	})
+	store := realtikvtest.CreateMockStoreAndSetup(t)
+	tk := createAsyncCommitTestKit(t, store)
+	tk.MustExec("drop table if exists t_parent, t_child")
+	tk.MustExec("create table t_parent (id int primary key)")
+	tk.MustExec("create table t_child (id int primary key, pid int, foreign key (pid) references t_parent(id) on delete cascade on update cascade)")
+	tk.MustExec("insert into t_parent values (1),(2),(3),(4)")
+	tk.MustExec("insert into t_child values (1,1),(2,2),(3,3)")
+	tk.MustExec("set tidb_enable_1pc = true")
+	tk.MustExec("begin pessimistic")
+	tk.MustExec("delete from t_parent where id in (1,4)")
+	tk.MustExec("update t_parent set id=22 where id=2")
+	tk.MustExec("commit")
+	tk.MustQuery("select * from t_parent order by id").Check(testkit.Rows("3", "22"))
+	tk.MustQuery("select * from t_child order by id").Check(testkit.Rows("2 22", "3 3"))
+}
+
+func TestTransactionIsolationAndForeignKey(t *testing.T) {
+	if !*realtikvtest.WithRealTiKV {
+		t.Skip("The test only support test with tikv.")
+	}
```

Comment:
We don't need this because all the tests in this package will be run with real TiKV.

Reply:
In the local environment, we can still run those tests with unistore. To avoid later being said that this test fails on unistore, I prefer keep this.